azure_pipelines/templates/tests-with-conf-file-mac.yml

```diff
@@ -0,0 +1,76 @@
+parameters:
```

Comment:
Why do we need a new template? can we use the existing one?

Reply:
I agree with @nilo-ms , it would better to avoid creating a similar script if possible.

You can see an example of an script with arguments in the `spm-integration-test.sh`. It is called from `pr-validation.yml`, and the name of the branch is passed as an argument.

Reply:
Indeed, the problem here is that the script itself has more lines, it's not about arguments but rather these extra 3 lines:
```
macos_version=$(sw_vers -productVersion)
echo "Mac Version $macos_version"      
full_path="build/Build/Products/${{ parameters.build }}_macosx$macos_version-x86_64.xctestrun"
```
I don't see a way to have them in the same script without parametrising the script itself

Reply:
Can you add the xcodebuild command in a parameter and use the iOS as default parameter value?

Reply:
Parametrised the `full_path`, `sdk`and `destination` parameters with defaults for MSAL UI tests